### ISSUE TITLE
Add post-clone-from-webdav reconfigure patch

### ISSUE THREAD

**Issue by mih** [datalad/datalad-next#108]
Cloning a dataset from a location that is shared through multiple accounts likely requires a reconfiguration of recorded annex special remotes pointing to the original account.

Use case: https://rdm.sfb1451.de/publication-catalog/#appendix-cloning-from-sciebo

We can add a patch that inspects the special remotes and reconfigures a webdav url to the one we have cloned from -- if the annex UUID matches.

**Comment by mslw** [comment]
> if the annex UUID matches

I looked into this issue, and it could be the specific situation I looked at (no annexed files in the special remote), or my poor understanding of annex, but I don't see the way to find a way how to check the annex UUID.

Starting with this in `remote.log` (I'm cutting out the actual url):
```
6090b605-84c5-4c16-b788-4764e14a9578 encryption=none exporttree=yes name=sciebo-storage type=webdav url=<THEIR URL> timestamp=1666180561.762543s
```
And trying to do:
```
git annex initremote my-proper-storage --private --sameas=sciebo-storage type=webdav exporttree=yes url=<MY URL>
```

I end up with:
```
6090b605-84c5-4c16-b788-4764e14a9578 encryption=none exporttree=yes name=sciebo-storage type=webdav url=<THEIR URL> timestamp=1666180561.762543s
ba2e76b3-9a39-41dd-a1d2-71f0b2879b62 exporttree=yes sameas-name=my-proper-storage sameas-uuid=6090b605-84c5-4c16-b788-4764e14a9578 type=webdav url=<MY URL> timestamp=1666976923.553779223s
```
This is what I wanted to achieve. But locally, I have two distinct UUIDs, and sameas annotation.

Thing is, this will happen and git-annex won't complain regardless if I point it to the right place, or a completely different place (i.e. special remote of some other repository on my server). So it doesn't seem to check, and I can't check by looking at the `remote.log` alone.

----

I was also thinking if and how to narrow down which special remotes should be tested. By URL? For sciebo, the URL is:
```
https://<SUBDOMAIN>.sciebo.de/remote.php/dav/files/<USER>/<PATH starting from user's home>
```
where subdomain, user, and possibly also leading elements of path will be different - so not much to grab onto other than sciebo domain itself.

**Comment by bpoldrack** [comment]
@mslw 
> git-annex won't complain regardless if I point it to the right place

Yes, annex can't. The point of `sameas` is, that it is a declaration. If annex could figure it out itself, there would be no need for `sameas`.

I'm also not sure what @mih suggests here. As in: What annex UUID is supposed to match what other. With RIA it works differently, because datalad is registering the special remote's UUID in the dataset's `config` in-store. So, postclone we can detect, that we have that special remote.

In any case, I'd like to register a thought here: We've had some trouble with the reconfiguration approach in the context of RIA/ORA. I want to remind, that datalad has the tooling to consider git's `url.....insteadOf` config. At least for some use cases utilizing this may be better than a reconfig of a special remote.
